### DISCUSSION TITLE
Vitest workspace run doesn't pick .env from individual projects

### DISCUSSION THREAD

**Discussion by semoal** [vitest-dev/vitest#5446] [Q&A] [answered]
I have a monorepo with different projects in `packages/*`, when we run `vitest` inside of this packages the .env.{mode} it's picked correctly, but when running in the root of the monorepo, 0 variables are caught. 

We have a `vitest.workspace.mts` file in the root with the following:

```mts
export default ["packages/*", "apps/*"];
```

We define the `vitest.config.ts` in the packages/ like this:
```ts
import tsconfigPaths from "vite-tsconfig-paths";
import { defineProject } from "vitest/config";

export default defineProject({
  envPrefix: ["SHERPAS_", "DATABASE_", "FLY_", "PRIMARY_"],
  plugins: [
    tsconfigPaths({
      root: "../..",
    }),
  ],
  test: {
    hookTimeout: 20_000,
    exclude: [
      "node_modules",
      "dist",
      ".idea",
      ".git",
      ".cache",
      "src/tests/**/*.{ts,tsx}",
    ],
    environment: "jsdom",
  },
});

```

What am I doing wrong?

**Comment by sheremet-va** [comment]
Can you provide more code? What is the directory structure? Maybe you can use this as an example? https://stackblitz.com/edit/vitest-dev-vitest-emzcxj?initialPath=__vitest__/

**Reply by semoal** [reply]
Yes, i prepare a repro :)

**Reply by semoal** [reply]
https://stackblitz.com/edit/vitest-dev-vitest-pxuer9?file=packages%2Flib1%2F.env.test
Here it is, if you run `npm test` from root, will fail the test, but if you run the `npm test` script inside the package lib1 will pass

**Reply by sheremet-va** [reply, marked as answer]
Thank you. Can confirm that it is a bug

**Reply by semoal** [reply]
Should I create an issue?

**Reply by sheremet-va** [reply]
Yes

**Reply by semoal** [reply]
Done: https://github.com/vitest-dev/vitest/issues/5448
Will close the discussion since it's already opened as bug :)
